mm2src/coins/eth/eth_swap_v2/eth_taker_swap_v2.rs

```diff
@@ -457,34 +462,105 @@ impl EthCoin {
         Ok(spend_payment_tx)
     }
 
-    /// Checks that taker payment state is `MakerSpent`.
-    /// Accepts maker spent payment transaction and returns it if payment status is correct.
-    pub(crate) async fn wait_for_taker_payment_spend_impl(
+    pub(crate) async fn find_taker_payment_spend_tx_impl(
```

Comment:
> let mut tx_hash: Option<H256> = None; can be removed and and_then used instead. It's not easy to review it in the current state.

I added it before loop in purpose. Once we find the transaction hash, we can skip re-fetching it in the next cycle iterations if smth go wrong. Its better to keep initialised variable before the loop to save found tx hash
```rust
        let mut tx_hash: Option<H256> = None;
        loop {
```


also introduced block windows. As dimxy correctly noticed, providers can have diff block limits when we request logs range https://github.com/KomodoPlatform/komodo-defi-framework/pull/2261#discussion_r1843775958 Thats why it could look complicated.
```rust
                // Split the range into windows of size logs_block_range
                while next_from_block <= current_block {
                    let to_block = std::cmp::min(next_from_block + self.logs_block_range - 1, current_block);
```

I will try to split the logic to reutilize functions in new and legacy code.

Reply:
@shamardy 

after splitting logic into separate functions, there is no need in option variable yes

https://github.com/KomodoPlatform/komodo-defi-framework/pull/2261/commits/e3b413225bfd87566987355992c4aa4f9eaeb4fc
https://github.com/KomodoPlatform/komodo-defi-framework/pull/2261/commits/9007f123183ef60ef71ddc6789cb42ae49f04caa